modules/core/02-client/keeper/params.go

```diff
@@ -6,19 +6,28 @@ import (
 	"github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
 )
 
-// GetAllowedClients retrieves the allowed clients from the paramstore
+// GetAllowedClients retrieves the allowed clients from the params stored.
 func (k Keeper) GetAllowedClients(ctx sdk.Context) []string {
-	var res []string
-	k.paramSpace.Get(ctx, types.KeyAllowedClients, &res)
-	return res
+	p := k.GetParams(ctx)
+	return p.AllowedClients
 }
 
 // GetParams returns the total set of ibc-client parameters.
 func (k Keeper) GetParams(ctx sdk.Context) types.Params {
-	return types.NewParams(k.GetAllowedClients(ctx)...)
+	var p types.Params
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get([]byte(types.ParamsKey))
+	if bz == nil {
+		return p
+	}
+
+	k.cdc.MustUnmarshal(bz, &p)
+	return p
```

Comment:
nit: I have a mild preference for consistency with other getter funcs

```suggestion
	store := ctx.KVStore(k.storeKey)
	bz := store.Get([]byte(types.ParamsKey))
	if bz == nil {
		return p
	}

        var params types.Params
	k.cdc.MustUnmarshal(bz, &params)
	return params
```

Reply:
I also don't really like named return args! 

It is, I thought I edited it before anyone saw 🙈

Reply:
could we apply this nit? i.e. move the var declaration below and rename `p` to `params`

Reply:
done, flew under the radar for some reason